stats/statsd/common/collector.go

```diff
@@ -0,0 +1,206 @@
+func (c *Collector) Link() string {
+	return c.Config.Addr.String
+}
+
+// Run the collector
+func (c *Collector) Run(ctx context.Context) {
+	c.Logger.Debugf("%s: Running!", c.Type.String())
+	ticker := time.NewTicker(pushInterval)
+	c.startTime = time.Now()
+
+	for {
+		select {
+		case <-ticker.C:
+			c.pushMetrics()
+		case <-ctx.Done():
+			c.pushMetrics()
+			c.finish()
+			return
+		}
+	}
+}
+
+// GetRequiredSystemTags Return the required system sample tags for the specific collector
+func (c *Collector) GetRequiredSystemTags() lib.TagSet {
+	return lib.TagSet{} // no tags are required
+}
+
+// SetRunStatus does nothing in statsd collector
+func (c *Collector) SetRunStatus(status lib.RunStatus) {}
+
+// Collect metrics
+func (c *Collector) Collect(containers []stats.SampleContainer) {
+	var pointSamples []*Sample
+
+	for _, container := range containers {
+		for _, sample := range container.GetSamples() {
+			pointSamples = append(pointSamples, generateDataPoint(sample))
+		}
+	}
+
+	if len(pointSamples) > 0 {
+		c.bufferLock.Lock()
+		c.buffer = append(c.buffer, pointSamples...)
+		c.bufferLock.Unlock()
+	}
+}
+
+func (c *Collector) pushMetrics() {
+	c.bufferLock.Lock()
+	if len(c.buffer) == 0 {
+		c.bufferLock.Unlock()
+		return
+	}
+	buffer := c.buffer
+	c.buffer = nil
+	c.bufferLock.Unlock()
+
+	c.Logger.
+		WithField("samples", len(buffer)).
+		Debugf("%s: Pushing metrics to server", c.Type.String())
+
+	if err := c.commit(buffer); err != nil {
+		c.Logger.
+			WithError(err).
+			Errorf("%s: Couldn't commit a batch", c.Type.String())
+	}
+}
+
+func (c *Collector) finish() {
+	if c.Type == Datadog {
+		c.sendSummaryData()
+	}
+	// Close when context is done
+	if err := c.Client.Close(); err != nil {
+		c.Logger.Debugf("%s: Error closing the client, %+v", c.Type.String(), err)
+	}
+}
+
+func (c *Collector) commit(data []*Sample) error {
+	for _, entry := range data {
+		c.dispatch(entry)
+	}
+	return c.Client.Flush()
+}
+
+func (c *Collector) dispatch(entry *Sample) {
+	var tagList []string
+	if c.FilterTags != nil {
+		tagList = c.FilterTags(entry.Data.Tags, entry.Extra.Group)
+	}
+	c.Summary[entry.Metric] = entry.Extra.Raw
+
+	switch entry.Type {
+	case stats.Counter:
+		_ = c.Client.Count(entry.Metric, int64(entry.Data.Value), tagList, 1)
+	case stats.Trend:
+		_ = c.Client.TimeInMilliseconds(entry.Metric, entry.Data.Value, tagList, 1)
+	case stats.Gauge:
+		_ = c.Client.Gauge(entry.Metric, entry.Data.Value, tagList, 1)
+	case stats.Rate:
+		if entry.Extra.Check != "" {
+			_ = c.Client.Count(
+				checkToString(entry.Extra.Check, entry.Data.Value),
+				1,
+				tagList,
+				1,
+			)
+		} else {
+			_ = c.Client.Count(entry.Metric, int64(entry.Data.Value), tagList, 1)
+		}
+	}
+}
+
+func checkToString(check string, value float64) string {
+	label := "pass"
+	if value == 0 {
+		label = "fail"
+	}
+	return fmt.Sprintf("check.%s.%s", check, label)
+}
+
+func (c *Collector) sendSummaryData() {
```

Comment:
I would like this to be behind an option actually. @robingustafsson  any opinion ? 
Also unfortunately it is currently with color sequences for shell which datadog doesn't understand :( 
So maybe just comment it out for now ?

Reply:
I vote for deleting this for now and making a new `enhancement` issue, since fixing/parameterizing the summary coloring code is getting a bit out of scope for this PR...

Reply:
Deleted for now

Reply:
added issue https://github.com/loadimpact/k6/issues/916